### ISSUE TITLE
Room Data Sheets - Error

### ISSUE THREAD

**Issue by cdfris** [dnenov/pyChilizer#17]
Receiving an error when attempting to run Room Data Sheets. 

Revit version 2023

pyRevit version 4.8.11.22151+0307-wip:277:2023.0.2

﻿

ERROR [pyrevit.revit.db.transaction] Error in Transaction Context. Rolling back changes. | <type 'exceptions.TypeError'>:FilterStringRule() takes exactly 3 arguments (4 given)

IronPython Traceback:
Traceback (most recent call last):
 File "C:\Users\Chris\AppData\Roaming\pyRevit\Extensions\pyChilizer.extension\pyChilizer.tab\Project.panel\rooms.stack\Room Data Sheets.pushbutton\script.py", line 195, in <module>
 File "C:\Users\Chris\AppData\Roaming\pyRevit\Extensions\pyChilizer.extension\lib\pychilizer\database.py", line 245, in unique_view_name
 File "C:\Users\Chris\AppData\Roaming\pyRevit\Extensions\pyChilizer.extension\lib\pychilizer\database.py", line 19, in get_view
 File "C:\Program Files\pyRevit-Master\pyrevitlib\pyrevit\revit\db\query.py", line 171, in get_biparam_stringequals_filter
TypeError: FilterStringRule() takes exactly 3 arguments (4 given)

Script Executor Traceback:
Microsoft.Scripting.ArgumentTypeException: FilterStringRule() takes exactly 3 arguments (4 given)
 at Microsoft.Scripting.Interpreter.ThrowInstruction.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.HandleException(InterpretedFrame frame, Exception exception)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet](T0 arg0, T1 arg1)
 at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
 at PyRevitLabs.PyRevit.Runtime.IronPythonEngine.Execute(ScriptRuntime& runtime)

**Comment by jmcouffin** [comment]
@cdfris 
you might want to try to run pyrevit with a different engine, it seems you are using engine 277.
It works on my install with engine 2710
to switch engine in command line (if your pyrevit clone is named with the default 'master', you can check your clone name running `pyrevit env` in cmd line:
`pyrevit attach master 2710 --installed --allusers`

**Comment by cdfris** [comment]
@jmcouffin 

Thanks for the suggestion - I changed the engine, but I am getting the same error.  Are you in Revit 2023?

**Comment by dnenov** [comment]
Hey @cdfris ! Thanks for the feedback, much appreciated. Indeed, this class has been changed in 2023 and we need to incorporate that change on our side. Please, bear with us while we handle this one.

**Comment by cdfris** [comment]
Thank you @dnenov !  This tool will be a huge time saver. Looking forward to using it.

**Comment by cdfris** [comment]
Hi @dnenov.  Sorry to bother, but do you have an idea of when this might be updated?

**Comment by dnenov** [comment]
Hey @cdfris ! Thank you for the kind nudge, just what I needed to activate myself and fix the issue. 

Actually, it turned out this is on pyRevit side. I have now created a PR for the pyRevit repository. When the moderator approves of it, the change will fix the issue you are experiencing with the Room Data Sheets extension. I can try to keep an eye for when that happens, I hope it is soon, and let you know!
